### ISSUE TITLE
Parser improvements: duration-based instead of timestamp-based

### ISSUE THREAD

**Issue by freethlua** [freethlua/individualogist-whats-your-archetype#38]
The new transcript lines that were re-worked upon earlier to make them shorter are much better but I found some lines that were very slightly out of sync. Re-syncing them may be a troublesome process. 

I was thinking of new format to be like this:

```
2000: Hello {{name}},
3000: If you’ve made it here, then congratulations!
3000: Because this means that your archetype is the Caregiver! {{#displayImage}}…{{/displayImage}}
3000: But what does this actually mean?
```

Notice that each line only has a duration associated with it (the number at the beginning), and not the time-stamp. 

So the first line `2000: Hello {{name}},` begins at 0 ms and holds for 2000 ms. And then the next line displays and holds for 3000 ms, and so on. 

This will make it much easier to adjust any line anywhere. 

Node: This will **not require re-writing** all transcripts either now nor probably in the future. A parser to convert between formats should be made. Also since there's probably not any transcription service that actually outputs in this format, any new transcripts will still need to be converted to this format, therefore a parser to convert between formats would also need to be created. 

This is only for facilitating minor adjustments in the timings of any transcript lines, so probably not a priority.

**Comment by zhixinlin** [comment]
Hi Freeth,

Just to be clear on what you've mentioned, I'm guessing we won't have to change the existing transcripts (txts) to the new format, but future scripts will have to be done in the new format. Is that right?

**Comment by freethlua** [comment]
Correct, and actually not even the (new) future scripts will have to be done in this format, the parser written for changing old script will work on new ones as well.

**Comment by zhixinlin** [comment]
That's wonderful. Let me know if you encounter any difficulties with doing this. How long do you think the new parsing will take? If it'll delay the completion by a lot, we can probably do without it for the time being.

**Comment by zhixinlin** [comment]
In any case, we're looking to run a few test ads to the new version coming Thursday or Friday. Will we need to migrate everything over to our server first?

**Comment by freethlua** [comment]
Sure, this is low priority. I'm actually working on #37 first which will solve #35 first. 

It is already on your server (archetype.individualogist.com), did you mean somewhere else?

**Comment by zhixinlin** [comment]
Oh I see that's right, my mistake. Nope we're good! Will wait on your update on the parsing before proceeding with testing. Thanks!

**Comment by freethlua** [comment]
Done. And kept both versions for now.
